src/http/index.js

```diff
@@ -405,6 +405,19 @@ export function encodeFormOrQuery(data) {
    * @param {string} parameterName - Parameter name
    * @return {object} encoded parameter names and values
    */
+  if (typeof data === 'string') {
+    try {
+      data = JSON.parse(data);
+      Object.entries(data).forEach(([key, value]) => {
+        if (typeof value === 'object' && !Array.isArray(value)) {
+          data[key] = JSON.stringify(value);
```

Comment:
Stringifying the nested value as otherwise it would be seen as undefined here: https://github.com/swagger-api/swagger-js/blob/ff9631fa905ecfd9fb7a92dad5c18455273404cd/src/http/index.js#L226
And if we did 
```js
data[key] = { value }
```
the value would be empty because of this: https://github.com/swagger-api/swagger-js/blob/ff9631fa905ecfd9fb7a92dad5c18455273404cd/src/http/index.js#L307

I think we could also use
```js
data[key] = { value, encoding: {} }
```
so that it's treated as an OpenAPI 3.x object but in the end it would just be stringified anyway:
https://github.com/swagger-api/swagger-js/blob/ff9631fa905ecfd9fb7a92dad5c18455273404cd/src/http/index.js#L285-L286